exercise_0/main.cpp

```diff
@@ -1,32 +1,29 @@
 
 #include <iostream>
-#include <cstring>
 #include <fstream>
 
-#define VOWELS  "aeiou"
+constexpr char VOWELS[] = "aeiou";
 
 
 bool    is_vowels(char character) {
-    char    lower;
-
-    lower = tolower(character);
-    for (int i = 0; VOWELS[i] != 0; i++) {
+    auto    lower = std::tolower(character);
+    for (int i = 0; VOWELS[i] != 0; ++i) {
         if (lower == VOWELS[i])
             return true;
     }
     return false;
 }
 
-void    count_words(char character, int *vowels, int *consonants, int *others) {
+void    count_words(char character, int &vowels, int &consonants, int &others) {
     if (!isalpha(character))
-        (*others)++;
+        others++;
     else if (is_vowels(character))
-        (*vowels)++;
+        vowels++;
     else
-        (*consonants)++;
+        consonants++;
 }
 
-int     words_checker(std::istream &read_file, char const *stop) {
+int     words_checker(std::istream &read_file, const std::string &stop = {}) {
```

Comment:
"words_checker" is good name for class, not for function. Widely spread convention said:
-function name should be "from verb"
-class name should be "from noun"